design-philosophy-and-threat-model.md

```diff
@@ -0,0 +1,128 @@
+1. Malicious external compromise of server infrastructure: In line with the principle of graceful system failures, we consider possible compromise of the server infrastructure, including of routers and load balancers. Where possible, the system should be robust up to some designated threshold of compromise. Mitigations include:
+    1. Infrastructure providers and compromised servers should not be able to mount effective brute force attacks on (partial) key material. Mitigations include:
+        1. The use of enclaves on servers: sensitive data and operations should take place on server enclaves or, optionally, on the asset owner’s device.
+        1. The use of threshold/MPC protocols to cryptographically enforce limits on the probability of a successful key recovery/key misuse attack by a subset of infrastructure providers/servers.
+    1. Periodic faults/failures of infrastructure should not render the system unusable. In particular, single points of failure should be avoided. This can be achieve by a system that allows for distribution of trust via flexible deployment across:
+        1. On-premises HSMs, if appropriate and desired.
+        1. Multiple providers of external cloud infrastructure, e.g., Microsoft Azure, AWS, GCP.
+        1. Use of multiple geographic/legal jurisdictions for infrastructure deployment.
+
+
+#### Infrastructure providers
+
+If the service provider enters into a partnership agreement with an infrastructure provider, e.g., AWS, GCP, Microsoft Azure, then we assume these entities are honest-but-curious.
+
+#### Other humans and malware
+
+Malware and device compromise is a consideration of Lock Keeper, particularly with respect to confidentiality assurance:
+
+1. Lock Keeper should provide confidentiality of the key and other sensitive user data on device and in transit:
+    1. A (whole) secret should never be held by any single entity, except possibly by the asset owner after an explicit, authorized request. Certain types of keys may not have full recovery functionality enabled, e.g., if the asset is considered a shared asset with the service provider. Threshold cryptography/MPC solutions should be leveraged to this end for key generation, distribution, and for signing/decryption.
+    1. The asset owner must authenticate in order to recover, use, or otherwise interact with their key. Care must be taken that the authentication mechanism is robust against impersonation attacks, e.g., by use of threshold cryptography/MPC solutions and/or a distributed PAKE.
+1. The developers integrating Lock Keeper should receive integration guidance on how to avoid decisions that undermine key or user data confidentiality.
+
+Humans with whom the digital asset owner interacts physically, e.g., family members, housemates, friends, and coffee shop dwellers, or virtually. Specifically, the software integrating Lock Keeper should include mitigations for the following types of threats:
```

Comment:
This looks like a bullet point but I think it's meant to be context for the following bullet points. Maybe rearrange the sentences something like this:

```suggestion
 The software integrating Lock Keeper should include mitigations against humans with whom the digital asset owner interacts physically, e.g., family members, housemates, friends, and coffee shop dwellers, or virtually. Specifically, it should consider the following types of threats:
```